### ISSUE TITLE
libprelude formula failed - target 'check'

### ISSUE THREAD

**Issue by black-snow** [Linuxbrew/homebrew-core#9512]
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single formula? 
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

> What you were trying to do (and why)

I was just `brew upgrade`'ing. Don't know which formula depends on `libprelude`.

> What happened (include command output)

```
==> Installing libprelude
nice: cannot set niceness: Permission denied
==> Downloading https://www.prelude-siem.org/attachments/download/721/libprelude-3.1.0.tar.gz
######################################################################## 100.0%
==> ./configure --prefix=/home/linuxbrew/.linuxbrew/Cellar/libprelude/3.1.0_2 --disable-rpath --with-pic --with-swig=yes --with-perl=no --with-python2=no --with-python3=no --with-valgrind=yes --with-lua=no -
==> make
==> make check
Last 15 lines from /home/xxx/.cache/Homebrew/Logs/libprelude/03.make:
make[5]: Leaving directory '/tmp/libprelude-20180921-8241-1og5jli/libprelude-3.1.0/libmissing/tests'
Makefile:3281: recipe for target 'check' failed
make[4]: *** [check] Error 2
make[4]: Leaving directory '/tmp/libprelude-20180921-8241-1og5jli/libprelude-3.1.0/libmissing/tests'
Makefile:1708: recipe for target 'check-recursive' failed
make[3]: *** [check-recursive] Error 1
make[3]: Leaving directory '/tmp/libprelude-20180921-8241-1og5jli/libprelude-3.1.0/libmissing'
Makefile:1896: recipe for target 'check' failed
make[2]: *** [check] Error 2
make[2]: Leaving directory '/tmp/libprelude-20180921-8241-1og5jli/libprelude-3.1.0/libmissing'
Makefile:1518: recipe for target 'check-recursive' failed
make[1]: *** [check-recursive] Error 1
make[1]: Leaving directory '/tmp/libprelude-20180921-8241-1og5jli/libprelude-3.1.0'
Makefile:1826: recipe for target 'check' failed
make: *** [check] Error 2

READ THIS: https://github.com/Linuxbrew/brew/wiki/troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core,
which support macOS only."
```

> What you expected to happen

Expected `brew upgrade` to succeed.

> Step-by-step reproduction instructions (by running `brew install` commands)

I didn't install `libprelude` myself. How do I find out which other formula depends on it?

**Comment by black-snow** [comment]
https://gist.github.com/black-snow/a58313b18983925487da6c7a34b750d0

**Comment by black-snow** [comment]
Errr... I ran upgrade again and it silently succeeded without doing anything. It should've done a pending rust upgrade and, I think, retried that libprelude.

**Comment by iMichka** [comment]
It may be a transient make check error in the build. If this is fixed for you, I propose that we close this issue.

**Comment by black-snow** [comment]
It's there again ;)

```
==> Pouring rust-1.29.1.x86_64_linux.bottle.tar.gz
nice: cannot set niceness: Permission denied
🍺  /home/linuxbrew/.linuxbrew/Cellar/rust/1.29.1: 46,587 files, 954.3MB
==> Reinstalling 2 broken dependents from source:
libprelude, rust
==> Reinstalling libprelude
nice: cannot set niceness: Permission denied
==> Downloading https://www.prelude-siem.org/attachments/download/721/libprelude-3.1.0.tar.gz
Already downloaded: /home/xxx/.cache/Homebrew/downloads/cc5fbb36642de2a026afe3cfe2705c5a17d21753d9638ffdad8b3fc75ca312e8--libprelude-3.1.0.tar.gz
==> ./configure --prefix=/home/linuxbrew/.linuxbrew/Cellar/libprelude/3.1.0_2 --disable-rpath --with-pic --with-swig=yes --with-perl=no --with-python2=no --with-
==> make
==> make check
Last 15 lines from /home/xxx/.cache/Homebrew/Logs/libprelude/03.make:
make[5]: Leaving directory '/tmp/libprelude-20180928-5746-5idghd/libprelude-3.1.0/libmissing/tests'
Makefile:3281: recipe for target 'check' failed
make[4]: *** [check] Error 2
make[4]: Leaving directory '/tmp/libprelude-20180928-5746-5idghd/libprelude-3.1.0/libmissing/tests'
Makefile:1708: recipe for target 'check-recursive' failed
make[3]: *** [check-recursive] Error 1
make[3]: Leaving directory '/tmp/libprelude-20180928-5746-5idghd/libprelude-3.1.0/libmissing'
Makefile:1896: recipe for target 'check' failed
make[2]: *** [check] Error 2
make[2]: Leaving directory '/tmp/libprelude-20180928-5746-5idghd/libprelude-3.1.0/libmissing'
Makefile:1518: recipe for target 'check-recursive' failed
make[1]: *** [check-recursive] Error 1
make[1]: Leaving directory '/tmp/libprelude-20180928-5746-5idghd/libprelude-3.1.0'
Makefile:1826: recipe for target 'check' failed
make: *** [check] Error 2
```

**Comment by iMichka** [comment]
```
Unconnected socket test... passed
Connected sockets test... failed (expecting POLLHUP after shutdown)
General socket test with fork... failed (expecting POLLHUP after shutdown)
```

**Comment by iMichka** [comment]
Looks similar to https://lists.gnu.org/archive/html/bug-gnulib/2018-05/msg00059.html

I would propose to disable that check (with a `if OS.mac?` condition). Would you like to open a PR to fix this?

**Comment by black-snow** [comment]
@iMichka Thanks for looking into this.

I'm not quite sure how we'd be able to do this. It's just `system "make", "check"`, isn't it? We'd have to change the Makefile, right? I'm not a c/make expert, maybe I'm missing something.

**Comment by iMichka** [comment]
We can disable the test by rewriting the libprelude.rb formula like this:

```
# https://lists.gnu.org/archive/html/bug-gnulib/2018-05/msg00059.html
system "make", "check" if OS.mac?
```

This is what I had in mind.

**Comment by iMichka** [comment]
Fixed by #10034
